### PR TITLE
Expose retry_budget within API DestinationRule ConnectionPoolSettings

### PR DESCRIPTION
**Pull request by ericdbishop**
Expose the Envoy CircuitBreaker retry_budget ConnectionPoolSettings within DestinationRule ConnectionPoolSettings HTTP configuration.

Credit to @quraynai as I used https://github.com/istio/api/pull/2734 as a reference.

Issue: https://github.com/istio/istio/issues/27419

### CODE REVIEW

**Comment by ericdbishop** [comment]
I provided more context in my PR in the main repo https://github.com/istio/istio/pull/51823

**Comment by zirain** [comment]
This was raised up again and again in the past, it's really meaningful.

Even knowing that this conflicts with an long timeexisting feature, I do think DestiontaionRule is the most appropriate place.

**Comment by ericdbishop** [comment]
@zirain we are discussing in https://github.com/istio/istio/pull/51823 if you'd like to join in the conversation.

File: kubernetes/customresourcedefinitions.gen.yaml [hunk 1]
```diff
@@ -413,6 +413,28 @@ spec:
...
+                                    requests.
+                                  properties:
+                                    budgetPercent:
+                                      description: The limit on concurrent retries
+                                        as a percentage of active requests and active
+                                        pending requests.
+                                      properties:
+                                        value:
+                                          format: double
+                                          type: number
```
Comment (ajvengo) [inline kubernetes/customresourcedefinitions.gen.yaml:428]: Why not add sane limits for percentage variable?
```suggestion
                                          type: number
                                          minimum: 0
                                          maximum: 100
```

File: kubernetes/customresourcedefinitions.gen.yaml [hunk 2]
```diff
@@ -629,6 +651,25 @@ spec:
...
+                            retryBudget:
+                              properties:
+                                budgetPercent:
+                                  description: The limit on concurrent retries as
+                                    a percentage of active requests and active pending
+                                    requests.
+                                  properties:
+                                    value:
+                                      format: double
+                                      type: number
```
Comment (ajvengo) [inline kubernetes/customresourcedefinitions.gen.yaml:663]: Same here:
```suggestion
                                      type: number
                                      minimum: 0
                                      maximum: 100
```

**Comment by keithmattix** [comment]
@istio/technical-oversight-committee Please review

**Comment by linsun** [comment]
@mikemorris will sync with @ericdbishop offline - we want to have this in gateway API first then implement in Istio.

**Pull request closed without merging**